File: packages/models/lib/src/units/unit.dart
import 'units.dart';

class Unit {

  const Unit({
    required this.unitKind,
    required this.name,
    required this.offense,
    required this.defenseInfantry,
    required this.defenseCavalry,
    required this.velocity,
    required this.cost,
    required this.upKeep,
    required this.time,
    required this.capacity,
    required this.infantry,
    required this.researchTime,
    required this.description,
  });
  final UnitKind unitKind;
  final String name;
  final int offense;
  final int defenseInfantry;
  final int defenseCavalry;
  final int velocity;
  final List<int> cost;
  final int upKeep;
  final int time;
  final int capacity;
  final bool infantry;
  final int researchTime;
  final String description;

  // For SPY units
  int getS() {
    return unitKind == UnitKind.SPY ? 35 : 0;
  }

  int getSD() {
    return unitKind == UnitKind.SPY ? 20 : 0;
  }
}

File: java_server/src/main/java/xyz/qruto/java_server/models/units/Unit.java
package xyz.qruto.java_server.models.units;

import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.List;

@Data
@AllArgsConstructor
public class Unit {
    private UnitKind unitKind;
    private String name;
    private int offense;
    private int defenseInfantry;
    private int defenseCavalry;
    private int velocity;
    private List<Integer> cost;
    private int upKeep;
    private int time;
    private int capacity;
    private boolean infantry;
    private int researchTime;
    private String description;

    //for SPY units
    public int getS(){
        return this.getUnitKind().equals(UnitKind.SPY) ? 35 : 0;
    }

    public int getSD(){
        return this.getUnitKind().equals(UnitKind.SPY) ? 20 : 0;
    }
}
